src/formatting/formatter/duration.rs

```diff
@@ -0,0 +1,183 @@
+            .iter()
+            .position(|&x| x == min_unit)
+            .error("min_unit must be one of \"y\", \"w\", \"d\", \"h\", \"m\", \"s\", or \"ms\"")?;
+
+        // UNITS are sorted largest to smallest
+        if min_unit_index < max_unit_index {
+            return Err(Error::new(format!(
+                "min_unit ({}) must be smaller than or equal to max_unit({})",
+                min_unit, max_unit,
+            )));
+        }
+
+        let units_upper_bound = min_unit_index - max_unit_index + 1;
+        let units = units.unwrap_or_else(|| min(units_upper_bound, 2));
+
+        if units > units_upper_bound {
+            return Err(Error::new(format!(
+                "there aren't {} units between min_unit ({}) and max_unit({})",
+                units, min_unit, max_unit,
+            )));
+        }
+
+        Ok(Self {
+            max_unit_index,
+            min_unit_index,
+            units,
+            round_up,
+            prefix_has_space,
+            pad_with,
+            show_leading_units_if_zero,
+        })
+    }
+}
+
+impl Formatter for DurationFormatter {
+    fn format(&self, val: &Value, _config: &SharedConfig) -> Result<String, FormatError> {
+        match val {
+            Value::Duration(duration) => {
+                let mut ms = duration.as_millis();
+                if self.round_up {
+                    ms += UNIT_CONVERSION_RATES[self.min_unit_index..]
+                        .iter()
+                        .product::<u128>()
+                        - 1;
+                }
+
+                let mut v = Vec::new();
+                let mut div = UNIT_CONVERSION_RATES[self.max_unit_index..=self.min_unit_index]
+                    .iter()
+                    .product::<u128>();
+                for rate in &UNIT_CONVERSION_RATES[self.max_unit_index..=self.min_unit_index] {
+                    v.push(ms / div);
+                    ms %= div;
+                    div /= rate;
+                }
+
+                let value_unit_seporator = if self.prefix_has_space { " " } else { "" };
+
+                Ok(itertools::join(
+                    v.iter()
+                        .enumerate()
+                        // Zip up the values of time with the matching units of time
+                        .zip(UNITS[self.max_unit_index..=self.min_unit_index].iter())
+                        // Skip wile the value is zero, unless we want to display the leading units of time with value of zero.
+                        // For example we want to have a minimum unit of seconds but to always show two values we could have:
+                        // " 0m 15s"
+                        .skip_while(|((i, value), _)| {
+                            **value == 0
+                                && (!self.show_leading_units_if_zero
+                                    || *i + self.max_unit_index
+                                        != self.min_unit_index - self.units + 1)
```

Comment:
```suggestion
                                        != self.min_unit_index + 1 - self.units)
```

because

```toml
[[block]]
block = "tea_timer"
format = " $time.dur(units:6) | click to start "
```

causes a panic:

```
thread 'main' panicked at src/formatting/formatter/duration.rs:163:44:
attempt to subtract with overflow
```